bazel/envoy_build_system.bzl

```diff
@@ -331,6 +363,7 @@ def envoy_proto_library(name, srcs = [], deps = [], external_deps = []):
         default_runtime = "@com_google_protobuf//:protobuf",
         protoc = "@com_google_protobuf//:protoc",
         deps = deps + cc_proto_deps,
+        linkstatic = 1,
```

Comment:
What's this `linkstatic=1` for? I believe for libraries it only prevents building a `.so`, it doesn't affect linking of cc_binaries that set static stuff through other means (i.e. envoy_cc_binary).

If the goal is to force symbols from these libraries to always be included in the final binary, use `alwayslink = 1`.

Reply:
It's exactly that, we don't want `.so`, it isn't needed in the Envoy build and there were `.so` generation issues with OSS-Fuzz injecting its fuzzing library (which is built completely outside of Bazel) into the link process.

Reply:
Got it. Could you put that into a comment?